src/application/user/getUser.usecase.ts

```diff
@@ -0,0 +1,15 @@
+import { UserEntity } from '../../domain/user/user.entity';
+import { UserRepository } from '../../domain/user/user.repository';
+
+export class GetUserUseCase {
+    constructor(private readonly userRepository: UserRepository) {}
+
+    public execute = async (
+        userId: string,
+    ): Promise<Partial<UserEntity> | null> => {
+        const user = await this.userRepository.getUserById(userId);
+        if (!user) return null;
+        const { password, ...userDetails } = user;
```

Comment:
This is better to do in a presenter.

Reply:
so we return the entire user?

Reply:
you should use a file called user.presenter.ts and there you must set which parameters you wan to show to upper layer. Don't transform userEntity directly in use case.